src/lv_draw/lv_draw_img.c

```diff
@@ -425,6 +425,41 @@ lv_img_src_t lv_img_src_get_type(const void * src)
     return img_src_type;
 }
 
+lv_img_dsc_t *lv_img_buf_alloc(lv_coord_t w, lv_coord_t h, lv_img_cf_t cf)
+{
+    /*Get the pixel size in bytes*/
+    uint8_t px_b = lv_img_color_format_get_px_size(cf) / 8;
+    
+    lv_img_dsc_t *dsc = lv_mem_alloc(sizeof(lv_img_dsc_t));
+    if(dsc == NULL)
+        return NULL;
+    
+    memset(dsc, 0, sizeof(lv_img_dsc_t));
+    dsc->data_size = px_b * w * h;
```

Comment:
It doesn't take into account the palette required for LV_IMG_CF_INDEXED images.

I suggest using these macros to determine the size: https://github.com/littlevgl/lvgl/blob/master/src/lv_objx/lv_canvas.h#L242

Unfortunately, these macros are tied to lv_canvas. They should be rather `LV_IMG_BUF_SIZE_...`.
What about adding `LV_IMG_BUF_SIZE_...` and use them in `LV_CANVAS_BUF_SIZE_...`?

Reply:
@kisvegabor I understand the issue. However, I don't understand how to use the canvas macros, given that any color format can be passed to this function.

Reply:
What about a `switch case` to use the proper macro?

Reply:
I see. I will try and implement this tonight.

Reply:
Implemented in https://github.com/littlevgl/lvgl/pull/1128/commits/9ad51e529e7711464a1934530809ab1454ef3da5.